### PR TITLE
[#23] Add import export of all configs

### LINKED ISSUE #23: Add Import/export of all configs
Add feature to import and export all configs to/from file.

### PR DESCRIPTION
**Pull request by mighantos**
Adds Export and Import of all configs to API.
Close #23

### CODE REVIEW

File: src/main/java/org/elasticsearch/app/api/server/services/ConfigManager.java [hunk 1]
```diff
@@ -122,6 +122,31 @@ public void addUpdateRecordToIndex(String indexName, UpdateRecord updateRecord)
...
+    public List<Map<String, Object>> getAllConfigs() {
+        return riverDAO.findAll().stream().map(River::toMap).collect(Collectors.toList());
+    }
+
+    @Transactional()
+    public List<Integer> setAllConfigs(List<String> configs) {
+        int newConfigs = 0;
+        int updatedConfigs = 0;
+        for (String config : configs) {
+            River newRiver = createRiver(config);
```
Comment (psiotwo) [inline src/main/java/org/elasticsearch/app/api/server/services/ConfigManager.java:145]: Isn't this in fact em.merge?

File: src/main/java/org/elasticsearch/app/api/server/IndexerController.java [hunk 2]
```diff
@@ -86,9 +85,18 @@ public void stopIndex(@PathVariable String id) {
...
     }
 
+    @PostMapping("/import/configs")
+    public List<Integer> importConfigs(@RequestBody List<String> jsonConfig) {
+        return configManager.setAllConfigs(jsonConfig);
+    }
 
+    @DeleteMapping("/configs/{id}")
+    public void deleteIndex(@PathVariable String id, @RequestParam(required = false, defaultValue = "false") boolean deleteData) {
+        River river = configManager.getRiver(id);
```
Comment (psiotwo) [inline src/main/java/org/elasticsearch/app/api/server/IndexerController.java]: Why is it necessary to retrieve from DB the whole object in order to delete it in the next row? Would not em.getReference be enough?

File: src/main/java/org/elasticsearch/app/api/server/services/ConfigManager.java [hunk 3]
```diff
@@ -122,6 +122,31 @@ public void addUpdateRecordToIndex(String indexName, UpdateRecord updateRecord)
...
+            River foundRiver = riverDAO.findById(newRiver.getRiverName()).orElse(null);
```
Comment (psiotwo) [inline src/main/java/org/elasticsearch/app/api/server/services/ConfigManager.java]: Not clear to me what is the interplay between id and rivername ...

**Review by psiotwo — COMMENTED** [review]
see comments

File: src/main/java/org/elasticsearch/app/api/server/IndexerController.java (hunk 2 above)
Comment (mighantos) [reply src/main/java/org/elasticsearch/app/api/server/IndexerController.java]: I am returning custom `ConfigNotFindException` when record with ID was not found => this prevents Elasticsearch exception. With lazy load I used `riverDAO.existsById(id)` but it returned true even when the record does not exist. So I landed on this.

File: src/main/java/org/elasticsearch/app/api/server/services/ConfigManager.java (hunk 3 above)
Comment (mighantos) [reply src/main/java/org/elasticsearch/app/api/server/services/ConfigManager.java]: riverName is the ID of obj River. Should i change it to ID?

File: src/main/java/org/elasticsearch/app/api/server/services/ConfigManager.java (hunk 1 above)
Comment (mighantos) [reply src/main/java/org/elasticsearch/app/api/server/services/ConfigManager.java:145]: It is. Last line in `for` does it `riverDAO.save(foundRiver);` But to retrieve how many was added as new I need to know if it exists => same problem as the next comment.

File: src/main/java/org/elasticsearch/app/api/server/IndexerController.java (hunk 2 above)
Comment (mighantos) [reply src/main/java/org/elasticsearch/app/api/server/IndexerController.java]: I will try to update SpringBoot found some similar bug description I older versions.

File: src/main/java/org/elasticsearch/app/api/server/services/ConfigManager.java (hunk 1 above)
Comment (mighantos) [reply src/main/java/org/elasticsearch/app/api/server/services/ConfigManager.java:145]: Changed to a reference. Merged by save().

File: src/main/java/org/elasticsearch/app/api/server/IndexerController.java (hunk 2 above)
Comment (mighantos) [reply src/main/java/org/elasticsearch/app/api/server/IndexerController.java]: Changed to reference with getById()

**Review by psiotwo — APPROVED** [review]
lgtm

**Pull request merged**